### ISSUE TITLE
push proceeds even if custom resource build fails

### ISSUE THREAD

**Issue by attilah** [aws-amplify/amplify-cli#8921]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14

### Amplify CLI Version

7.x

### What operating system are you using?

macos

### Amplify Categories

Not applicable

### Amplify Commands

push

### Describe the bug

When the typescript compilation fails for custom resources push still proceed (and fail)

### Expected behavior

Don't proceed with push if compilation failed

### Reproduction steps

1. add custom resource
2. modify the TS file to not to compile
3. push

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by kaustavghosh06** [comment]
Yes. It shows the error in red with compilation and then asks 'Do you want to proceed?
<img width="1124" alt="Screen Shot 2021-11-17 at 11 06 23 AM" src="https://user-images.githubusercontent.com/7247658/142266471-30338db1-9d13-4083-9b04-d4613f2923e7.png">
"

**Comment by attilah** [comment]
And on --yes it proceeds

**Comment by garciajg** [comment]
I am also having the same issue

**Comment by sobolk** [comment]
This has been fixed in `@aws-amplify/cli@11.0.0`
